src/ItemInputCharset.h

```diff
@@ -0,0 +1,64 @@
+#ifndef ItemInputCharset_H
+#define ItemInputCharset_H
+
+#include <utils/utils.h>
+
+#include "ItemInput.h"
+
+class ItemInputCharset : public ItemInput {
+   private:
+    const char* charset;
+    const uint8_t charsetSize;
+    // Active index of the charset
+    int8_t charsetPosition = -1;
+
+    void changeChar(const char character) {
+        uint8_t length = strlen(value);
+        if (cursor < length) {
+            value[cursor] = character;
+        } else {
+            char* buf = new char[length + 2];
+            concat(value, character, buf);
+            value = buf;
+        }
+        MenuItem::draw();
+        display->resetBlinker(constrainBlinkerPosition(display->getBlinkerPosition()));
+        // Log
+        printCmd(F("TYPE-CHARSET"), character);
+    }
```

Comment:
The value of the text shouldn't be set immediately.

From the video I posted, you can see that I scroll through the charset and then press `enter` to confirm and that's when the value is set and the cursor moves to the next column.
If I don't confirm and exit (go back), that value will not be persisted.

To achieve this, you'll need `override enter()` too

Reply:
Ok, I see. What about navigation over the string (`left` and `right` actions)?
Should they behave as `back` and rollback changes?
Or it can be like 2 states - navigate over the string and change char. And each state listen only it's own pair `left` and `right` for navigation, `up` and `down` for change char. And states are separated by `enter`/`back` actions.

P.S. How did you create so nice videos? Is it some simulation?

Reply:
Also I think it can be 2 different implementation for the same item. Not right now, but it's possible.
Because Rotary encoder has not enough dimensions - clockwise rotation can be both `righ` and `down`.
But keyboard can have all the actions simultaneously.

Reply:
> Should they behave as back and rollback changes?

No, it behaves just as a PC would behave in Notepad for example.
When `left` and `right` is called, it just moves the cursor to that position and you can then use `up` and `down` to go through the charset and execute `enter` to persist it, you can as well execute `backspace` to remove that specific char where the cursor is.

> Also I think it can be 2 different implementation for the same item. Not right now, but it's possible.
Because Rotary encoder has not enough dimensions - clockwise rotation can be both righ and down.
But keyboard can have all the actions simultaneously.

Yes, right now rotary encode can only do `up` and `down` when in edit mode, I will brainstorm if there's a way to support `left` and `right` movement

> P.S. How did you create so nice videos? Is it some simulation?

I use wokwi, you can install the plugin in VS Code. Create a `main.cpp` file in src should contain the code you want to run, compile the project then open wokwi.
If you compile with PlatformIO, it should work well, see the `wokwi.toml` file in the project

Reply:
Hmm.. I think they shouldn't be skipped. I don't remember how it behaves currently.

You enter edit mode
Cursor is placed at the end
```
┌───────────────┐
| A:  A B C █ 
└───────────────┘
```
You start picking a char (up/down)
```
┌───────────────┐
| A:  V V V ⍌
└───────────────┘
```
You go left (since enter was not pressed the value is wiped, restored to previous state which is empty in this case)
```
┌───────────────┐
| A:  V V ⍌ 
└───────────────┘

I hope it's understandable.

You can always move left and right and edit the value but you have to press enter to persist it as soon as you press left or right without persisting, the value will be restored to previous value or empty vale if there's no previous value at that position.

In terms of code, after moving left/right, you will try to get the char at the `blinkerPosition +/- 1` (the previousPosition) in the value of the input. If it is null, draw a blank character else draw the char at that position in the value.
When you press enter, the value of the input is adjusted.
- if the cursor is at the start you take the desired character and append the rest of the input minus the start character.
- if the cursor is at the end you take the input value minus the last character and append the desired character.
- if the cursor is within the value you take the value up to but excluding that position and append the desired character then append the rest of the value

There are helpers for this in the utils
- `concat(const char* first, char second, const char* third, char* result)`
- `concat(const char* first, char second, char* result)`
- etc

What do you think @ShishkinDmitriy?, I believe it behaves this way in `master` the only missing part is the redrawing of the value on `left/right`

Reply:
I just tried `master` version. Looks like it skips `left` and `right`. Also big strings are not correctly rendered - only first letter.

Anyway I did as you described.

- `left` and `right` are not skipped, they just rollback any changes made with char. 
- Visible area moved when overflow. 
- Also when you initially change the value it tries to search position in charset and tries to follow. Master version always start with 0.